operator/controllers/execution/scheduledscan_controller_test.go

```diff
@@ -1,63 +1,106 @@
-		},
-		{
-			in: map[string]string{
-				"barfoo.securecodebox.io/bar": "bar",
-				"barfoo.securecodebox.io/foo": "bar",
+			{
+				in: map[string]string{
+					"barfoo.securecodebox.io/bar": "bar",
+					"foo":                         "bar",
+				},
+				expectedMapKeyLength: 1,
 			},
-			expectedMapKeyLength: 2,
-		},
-	}
-
-	for _, test := range tests {
-		scheduledScan := executionv1.ScheduledScan{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:        "foobar",
-				Annotations: test.in,
+			{
+				in: map[string]string{
+					"barfoo.securecodebox.io/bar": "bar",
+					"barfoo.securecodebox.io/foo": "bar",
+				},
+				expectedMapKeyLength: 2,
 			},
 		}
-		actual := getAnnotationsForScan(scheduledScan)
-		if len(actual) != test.expectedMapKeyLength {
-			t.Error(fmt.Errorf("getAnnotationsForScan should only copy over annotations following the pattern '*.securecodebox.io', but map: %v returned a map with %d keys (%d expected)", test.in, len(actual), test.expectedMapKeyLength))
-		}
-	}
-}
+		It("Should drop all annotations not prefixed with \"*.securecodebox.io/*\"", func() {
+			for _, test := range tests {
+				scheduledScan := executionv1.ScheduledScan{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        "foobar",
+						Annotations: test.in,
+					},
+				}
+				actual := getAnnotationsForScan(scheduledScan)
+				Expect(len(actual)).Should(Equal(test.expectedMapKeyLength))
+			}
+		})
+	})
+	Context("A Scan is triggred due to a Scheduled Scan", func() {
+		It("The ScheduledScan's Finding Summary shoud be updated of with the results of the successful Scan", func() {
+			ctx := context.Background()
+			namespace := "scantype-multiple-scheduled-scan-triggerd-test"
+
+			createNamespace(ctx, namespace)
+			createScanType(ctx, namespace)
+			scheduledScan := createScheduledScan(ctx, namespace, true)
+
+			var scanlist executionv1.ScanList
+			// ensure that the ScheduledScan has been triggered
+			waitForScheduledScanToBeTriggered(ctx, namespace)
+			k8sClient.List(ctx, &scanlist, client.InNamespace(namespace))
+
+			Expect(scanlist.Items).Should(HaveLen(1))
+
+			scan := scanlist.Items[0]
+			scan.Status.State = "Done"
+
+			scan.Status.Findings = executionv1.FindingStats{
+				Count:             42,
+				FindingSeverities: executionv1.FindingSeverities{High: 42},
+				FindingCategories: map[string]uint64{"Open Port": 42},
+			}
+
+			k8sClient.Status().Update(ctx, &scan)
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "test-scan", Namespace: namespace}, &scheduledScan)
+				if errors.IsNotFound(err) {
+					panic("ScheduledScan should be present for this check!")
+				}
+				return (scheduledScan.Status.Findings.Count == 42 &&
+					scheduledScan.Status.Findings.FindingSeverities == executionv1.FindingSeverities{High: 42} &&
+					reflect.DeepEqual(scheduledScan.Status.Findings.FindingCategories, map[string]uint64{"Open Port": 42}))
+			}, timeout, interval).Should(BeTrue())
```

Comment:
I'd try to use the Ginko Assertions as much as possible as these will print out a more clear diff when something goes wrong. Something like this:

(Suggestion can't be applied directly as this also removes the need for the reflect import)

```suggestion
			Eventually(func() bool {
				err := k8sClient.Get(ctx, types.NamespacedName{Name: "test-scan", Namespace: namespace}, &scheduledScan)
				if errors.IsNotFound(err) {
					panic("ScheduledScan should be present for this check!")
				}
				return scheduledScan.Status.Findings.Count != 0
			}, timeout, interval).Should(BeTrue())

			Expect(scheduledScan.Status.Findings.Count).Should(Equal(uint64(42)))
			Expect(scheduledScan.Status.Findings.FindingSeverities).Should(Equal(executionv1.FindingSeverities{High: 42}))
			Expect(scheduledScan.Status.Findings.FindingCategories).Should(Equal(map[string]uint64{"Open Port": 42}))
```